Firestore/core/src/firebase/firestore/remote/datastore.h

```diff
@@ -70,18 +101,28 @@ class Datastore {
  private:
   void PollGrpcQueue();
 
+  using OnToken = std::function<void(const auth::Token&)>;
+  using OnError = std::function<void(const util::Status&)>;
+  void WithToken(const OnToken& on_token, const OnError& on_error);
```

Comment:
This is called `invokeRPCWithFactory: errorHandler:` [in Objective-C](https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Source/Remote/FSTDatastore.mm#L317-L334). Since it doesn't exist on other platforms, I think diverging is okay.